src/config.js

```diff
@@ -77,8 +78,23 @@ export const settings = {
     description: 'Enable or disable all new feature hints in the interface.',
     default: true,
   },
+  showDatasetPreviews: {
+    name: 'Dataset previews',
+    description: 'Display preview data for all datasets.',
+    default: true,
+  },
 };
 
+// Fetch preferences from the backend and update settings
+(async () => {
+  try {
+    const preferences = await fetchPreferences();
+    settings.showDatasetPreviews.default = preferences.showDatasetPreviews;
+  } catch (error) {
+    console.error('Error fetching initial preferences:', error);
+  }
+})();
```

Comment:
@SajidAlamQB I think this might not be the right place to make call, as its async, it may or may not update Redux state, I think you should explore redux-thunk.

Reply:
@SajidAlamQB, As I was thinking the api call can be delayed and it can be called when someone clicks on the settings icon (before we open the settings modal). That way you can dispatch an action when someone clicks on settings icon -> get the preferences json -> update the toggle accordingly -> open the modal popup which has the updated state. The other way is to call the api along with /main on first load, similar to /api/package-compatibilities. 

This file can be left with the default as before. We will only update redux store.